transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoOps.java

```diff
@@ -19,137 +19,185 @@
+     *              __u32 __pad1;
+     *          };
+     *      };
+     *
+     *      union {             // union2
+     *          __u64 addr;    // pointer to buffer or iovecs
+     *          __u64 splice_off_in;
+     *          struct {
+     *              __u32 level;
+     *              __u32 optname;
+     *          };
+     *      };
+     *      __u32 len;        // buffer size or number of iovecs
+     *
+     *      union {           // union3
+     *          __kernel_rwf_t rw_flags;
+     *          __u32 fsync_flags;
+     *          __u16 poll_events;    // compatibility
+     *          __u32 poll32_events; // word-reversed for BE
+     *          __u32 sync_range_flags;
+     *          __u32 msg_flags;
+     *          __u32 timeout_flags;
+     *          __u32 accept_flags;
+     *          __u32 cancel_flags;
+     *          __u32 open_flags;
+     *          __u32 statx_flags;
+     *          __u32 fadvise_advice;
+     *          __u32 splice_flags;
+     *          __u32 rename_flags;
+     *          __u32 unlink_flags;
+     *          __u32 hardlink_flags;
+     *          __u32 xeattr_flags;
+     *          __u32 msg_ring_flags;
+     *          __u32 uring_cmd_flags;
+     *          __u32 waitid_flags;
+     *          __u32 futex_flags;
+     *          __u32 install_fd_flags;
+     *          __u32 nop_flags;
+     *      };
+     *      __u64 user_data;    // data to be passed back at completion time
+     *                          // pack this to avoid bogus arm OABI complaints
+     *
+     *      union {             // union4
+     *
+     *          // index into fixed buffers, if used
+     *          __u16 buf_index;
+     *          // for grouped buffer selection
+     *          __u16 buf_group;
+     *      }__attribute__((packed));
+     *      // personality to use, if used
+     *      __u16 personality;
+     *
+     *      union {            // union5
+     *
+     *          __s32 splice_fd_in;
+     *          __u32 file_index;
+     *          __u32 optlen;
+     *          struct {
+     *              __u16 addr_len;
+     *              __u16 __pad3[ 1];
+     *          };
+     *      };
+     *
+     *      union {           // union6
+     *
+     *          struct {
+     *              __u64 addr3;
+     *              __u64 __pad2[ 1];
+     *          };
+     *          __u64 optval;
+     *          //
+     *          // If the ring is initialized with IORING_SETUP_SQE128, then
+     *          // this field is used for 80 bytes of arbitrary command data
+     *          __u8 cmd[ 0];
+     *      };
+     *  };
+     * }
+     * </pre>
      */
-    public IoUringIoOps(byte opcode, int flags, short ioPrio, int fd, int rwFlags, long bufferAddress,
-                        int length, long offset, int spliceFdIn, short data) {
+    public IoUringIoOps(byte opcode, byte flags, short ioPrio, int fd, long union1, long union2, int len, int union3,
```

Comment:
We might want to consider adding a builder later to make things easier.

Reply:
Yeah, a builder could have overloads for the unions, so you can set the field with the right name for the context.

Reply:
lets do this as a followup tho